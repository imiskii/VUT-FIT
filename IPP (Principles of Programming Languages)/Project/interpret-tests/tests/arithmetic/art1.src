.ippcode23

DEFVAR GF@a
DEFVAR GF@b
DEFVAR GF@c
DEFVAR GF@d

MOVE GF@a int@10
MOVE GF@b int@-2
MOVE GF@c int@0
MOVE GF@d string@5


DEFVAR GF@x

ADD GF@x GF@a GF@b
WRITE GF@x
WRITE string@|

ADD GF@x int@10 GF@a
WRITE GF@x
WRITE string@|

ADD GF@x GF@a int@10 
WRITE GF@x
WRITE string@|

ADD GF@x int@50 int@10 
WRITE GF@x
WRITE string@||
###


SUB GF@x GF@a GF@b
WRITE GF@x
WRITE string@|

SUB GF@x int@10 GF@a
WRITE GF@x
WRITE string@|

SUB GF@x GF@a int@10 
WRITE GF@x
WRITE string@|

SUB GF@x int@50 int@10 
WRITE GF@x
WRITE string@||

###

MUL GF@x GF@a GF@b
WRITE GF@x
WRITE string@|

MUL GF@x int@50 GF@b
WRITE GF@x
WRITE string@|

MUL GF@x GF@b int@50
WRITE GF@x
WRITE string@|

MUL GF@x int@-2 int@50
WRITE GF@x
WRITE string@||

###

IDIv GF@x GF@a GF@b
WRITE GF@x
WRITE string@|

IDIv GF@x GF@a int@10
WRITE GF@x
WRITE string@|

IDIv GF@x int@10 GF@a
WRITE GF@x
WRITE string@|

IDIv GF@x int@-5 int@-2
WRITE GF@x
WRITE string@||