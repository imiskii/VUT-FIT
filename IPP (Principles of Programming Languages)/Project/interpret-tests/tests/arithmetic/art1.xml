<?xml version="1.0" encoding="UTF-8"?>
<program language="IPPcode23">
 <instruction order="1" opcode="DEFVAR">
  <arg1 type="var">GF@a</arg1>
 </instruction>
 <instruction order="2" opcode="DEFVAR">
  <arg1 type="var">GF@b</arg1>
 </instruction>
 <instruction order="3" opcode="DEFVAR">
  <arg1 type="var">GF@c</arg1>
 </instruction>
 <instruction order="4" opcode="DEFVAR">
  <arg1 type="var">GF@d</arg1>
 </instruction>
 <instruction order="5" opcode="MOVE">
  <arg1 type="var">GF@a</arg1>
  <arg2 type="int">10</arg2>
 </instruction>
 <instruction order="6" opcode="MOVE">
  <arg1 type="var">GF@b</arg1>
  <arg2 type="int">-2</arg2>
 </instruction>
 <instruction order="7" opcode="MOVE">
  <arg1 type="var">GF@c</arg1>
  <arg2 type="int">0</arg2>
 </instruction>
 <instruction order="8" opcode="MOVE">
  <arg1 type="var">GF@d</arg1>
  <arg2 type="string">5</arg2>
 </instruction>
 <instruction order="9" opcode="DEFVAR">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="10" opcode="ADD">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="var">GF@a</arg2>
  <arg3 type="var">GF@b</arg3>
 </instruction>
 <instruction order="11" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="12" opcode="WRITE">
  <arg1 type="string">|</arg1>
 </instruction>
 <instruction order="13" opcode="ADD">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="int">10</arg2>
  <arg3 type="var">GF@a</arg3>
 </instruction>
 <instruction order="14" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="15" opcode="WRITE">
  <arg1 type="string">|</arg1>
 </instruction>
 <instruction order="16" opcode="ADD">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="var">GF@a</arg2>
  <arg3 type="int">10</arg3>
 </instruction>
 <instruction order="17" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="18" opcode="WRITE">
  <arg1 type="string">|</arg1>
 </instruction>
 <instruction order="19" opcode="ADD">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="int">50</arg2>
  <arg3 type="int">10</arg3>
 </instruction>
 <instruction order="20" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="21" opcode="WRITE">
  <arg1 type="string">||</arg1>
 </instruction>
 <instruction order="22" opcode="SUB">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="var">GF@a</arg2>
  <arg3 type="var">GF@b</arg3>
 </instruction>
 <instruction order="23" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="24" opcode="WRITE">
  <arg1 type="string">|</arg1>
 </instruction>
 <instruction order="25" opcode="SUB">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="int">10</arg2>
  <arg3 type="var">GF@a</arg3>
 </instruction>
 <instruction order="26" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="27" opcode="WRITE">
  <arg1 type="string">|</arg1>
 </instruction>
 <instruction order="28" opcode="SUB">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="var">GF@a</arg2>
  <arg3 type="int">10</arg3>
 </instruction>
 <instruction order="29" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="30" opcode="WRITE">
  <arg1 type="string">|</arg1>
 </instruction>
 <instruction order="31" opcode="SUB">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="int">50</arg2>
  <arg3 type="int">10</arg3>
 </instruction>
 <instruction order="32" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="33" opcode="WRITE">
  <arg1 type="string">||</arg1>
 </instruction>
 <instruction order="34" opcode="MUL">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="var">GF@a</arg2>
  <arg3 type="var">GF@b</arg3>
 </instruction>
 <instruction order="35" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="36" opcode="WRITE">
  <arg1 type="string">|</arg1>
 </instruction>
 <instruction order="37" opcode="MUL">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="int">50</arg2>
  <arg3 type="var">GF@b</arg3>
 </instruction>
 <instruction order="38" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="39" opcode="WRITE">
  <arg1 type="string">|</arg1>
 </instruction>
 <instruction order="40" opcode="MUL">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="var">GF@b</arg2>
  <arg3 type="int">50</arg3>
 </instruction>
 <instruction order="41" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="42" opcode="WRITE">
  <arg1 type="string">|</arg1>
 </instruction>
 <instruction order="43" opcode="MUL">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="int">-2</arg2>
  <arg3 type="int">50</arg3>
 </instruction>
 <instruction order="44" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="45" opcode="WRITE">
  <arg1 type="string">||</arg1>
 </instruction>
 <instruction order="46" opcode="IDIV">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="var">GF@a</arg2>
  <arg3 type="var">GF@b</arg3>
 </instruction>
 <instruction order="47" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="48" opcode="WRITE">
  <arg1 type="string">|</arg1>
 </instruction>
 <instruction order="49" opcode="IDIV">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="var">GF@a</arg2>
  <arg3 type="int">10</arg3>
 </instruction>
 <instruction order="50" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="51" opcode="WRITE">
  <arg1 type="string">|</arg1>
 </instruction>
 <instruction order="52" opcode="IDIV">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="int">10</arg2>
  <arg3 type="var">GF@a</arg3>
 </instruction>
 <instruction order="53" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="54" opcode="WRITE">
  <arg1 type="string">|</arg1>
 </instruction>
 <instruction order="55" opcode="IDIV">
  <arg1 type="var">GF@x</arg1>
  <arg2 type="int">-5</arg2>
  <arg3 type="int">-2</arg3>
 </instruction>
 <instruction order="56" opcode="WRITE">
  <arg1 type="var">GF@x</arg1>
 </instruction>
 <instruction order="57" opcode="WRITE">
  <arg1 type="string">||</arg1>
 </instruction>
</program>
