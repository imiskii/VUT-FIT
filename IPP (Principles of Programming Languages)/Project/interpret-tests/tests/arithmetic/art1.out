8|20|20|60||12|0|0|40||-20|-100|-100|-100||-5.0|1.0|1.0|2.5||